### PR TITLE
Use r2 for all endpoints with extensions.

### PR DESCRIPTION
**Pull request by wting**
:eyeglasses: @prashtx @uzi

### CODE REVIEW

File: default.vcl [hunk 1]
```diff
@@ -20,6 +20,10 @@ sub vcl_recv {
       # Blacklisted endpoints
       # This endpoint 200's for mweb, but the current implementation has issues
       req.url !~ "^/message/compose/?$"
+      # We don't host json in mweb
+      || req.url !~ ".*\.json(\?.*|$)"
+      # We don't host XML / RSS in mweb
+      || req.url !~ "(?i).*\.(xml|rss)$"
```
Comment (prashtx) [inline default.vcl]: If `req.url` contains query params in addition to the path, then I think we need the same `(\?.*|$)` handling for xml/rss that you have for json (we support `?limit=50`, for example).

**Comment by wting** [comment]
💇

**Comment by prashtx** [comment]
:fish:

**Comment by wting** [comment]
💇

**Comment by wting** [comment]
Here's a handy [visual aid](https://regexper.com/#%5C%2F.%2B%5C.%5Ba-z%5D%7B3%2C4%7D(%5C%3F.*%7C%24)) for parsing the regex. Keep in mind that link has an off by one error.

**Comment by prashtx** [comment]
:fish:

**Pull request merged**